ngraph/core/include/ngraph/op/util/fft_base.hpp

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "ngraph/op/op.hpp"
+#include "ngraph/op/util/attr_types.hpp"
+
+namespace ngraph
+{
+    namespace op
+    {
+        namespace util
+        {
+            /// \brief Abstract base class for operations DFT and DFT.
+            class NGRAPH_API FFTBase : public Op
+            {
+            public:
+                NGRAPH_RTTI_DECLARATION;
+                FFTBase() = default;
+
+                void validate_and_infer_types() override;
+                bool visit_attributes(AttributeVisitor& visitor) override;
+
+            protected:
+                /// \brief Constructs an FFT operation. FFT is performed for full size axes.
+                ///
+                /// \param data  Input data
+                /// \param axes Axes to perform FFT
+                FFTBase(const Output<Node>& data, const Output<Node>& axes);
```

Comment:
Because we needn't to create `FFTBase` instances, but only `DFT` or `IDFT` ones. I wrote `FFTBase` class, using `BinaryElementwiseArithmetic` class as a reference.

Reply:
I think you can also can move the default constructor to protected.